pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -276,17 +271,21 @@ func (l *podNIC) PlugPhase2(domain *api.Domain) error {
 		return err
 	}
 
-	if err = bindMechanism.loadCachedDhcpConfig(); err != nil {
-		log.Log.Reason(err).Critical("failed to load cached vif configuration")
-	}
-
 	if err := bindMechanism.decorateConfig(*domainIface); err != nil {
 		log.Log.Reason(err).Critical("failed to create libvirt configuration")
 	}
 
-	if err := ensureDHCP(bindMechanism, l.podInterfaceName); err != nil {
-		log.Log.Reason(err).Criticalf("failed to ensure dhcp service running for %s: %s", l.podInterfaceName, err)
-		panic(err)
+	if l.dhcpConfigurator != nil {
+		dhcpConfig, err := l.dhcpConfigurator.ImportConfiguration(l.podInterfaceName)
+		if err != nil || dhcpConfig == nil {
+			log.Log.Reason(err).Critical("failed to load cached dhcpConfig configuration")
+		}
+		if !dhcpConfig.IPAMDisabled {
+			if err := l.dhcpConfigurator.EnsureDhcp(l.podInterfaceName, *dhcpConfig, l.iface.DHCPOptions); err != nil {
+				log.Log.Reason(err).Criticalf("failed to ensure dhcp service running for: %s", l.podInterfaceName)
+				panic(err)
+			}
+		}
```

Comment:
All this can be done inside the DhcpConfigurator, so we mov more logic under dhcp package.